### PR TITLE
gh-110264: Handle `PyList_Reverse` error in `posixmodule.c`

### PR DESCRIPTION
**Pull request by sobolevn**
As I said, I am sorry! 😆 

* Issue: gh-110262

* Issue: gh-110264

### CODE REVIEW

File: Modules/posixmodule.c [hunk 1]
```diff
@@ -571,8 +571,12 @@ run_at_forkers(PyObject *lst, int reverse)
         if (cpy == NULL)
             PyErr_WriteUnraisable(lst);
         else {
-            if (reverse)
-                PyList_Reverse(cpy);
+            if (reverse) {
+                if (PyList_Reverse(cpy) < 0) {
+                    Py_DECREF(cpy);
+                    return;
```
Comment (serhiy-storchaka) [inline Modules/posixmodule.c:577]: Add `PyErr_WriteUnraisable()`.

**Comment by serhiy-storchaka** [comment]
Actually, `PyList_Reverse()` only fails if the argument is not a list. But it is a list here, so it never fails.

**Comment by encukou** [comment]
That's the *current* implementation of `PyList_Reverse`. If the implementation changes, will we remember to update the call site?

**Comment by serhiy-storchaka** [comment]
It has not changed in the last 25 years. If it suddenly changes, I'm sure we'll have a long discussion before that.

**Pull request closed without merging**